_ingest-pipelines/processors/ml-inference.md

```diff
@@ -150,9 +194,263 @@ The response confirms that, in addition to the `passage_text` field, the process
 Once you have created an ingest pipeline, you need to create an index for ingestion and ingest documents into the index.
 {: .note}
 
----
+### Example: Local model
+
+The following example configures an `ml_inference` processor with a local model.
+
+**Step 1: Create a pipeline**
+
+The following example creates an ingest pipeline for the `huggingface/sentence-transformers/all-distilroberta-v1` local model. The model is a sentence transformer [pretrained model]({{site.url}}{{site.baseurl}}/ml-commons-plugin/pretrained-models/#sentence-transformers) hosted in your OpenSearch cluster. 
+
+If you invoke the model using the Predict API, the request looks like this:
+
+```json
+POST /_plugins/_ml/_predict/text_embedding/cleMb4kBJ1eYAeTMFFg4
+{
+  "text_docs":[ "today is sunny"],
+  "return_number": true,
+  "target_response": ["sentence_embedding"]
+}
+```
+
+Using this schema, you'll specify the `model_input` as follows:
+
+```json
+ "model_input": "{ \"text_docs\": ${input_map.text_docs}, \"return_number\": ${model_config.return_number}, \"target_response\": ${model_config.target_response} }"
+```
+
+The model converts the text in the `text_docs` field to embeddings so it expects an ingested document to contain a `text_docs` field. In this example, ingested documents do not contain a `text_docs` field. The `model_input` specifies to source the `text_docs` field from the `input_map`. The `input_map`, in turn, specifies that the input text resides at the `book.*.chunk.text.*.context` JSON path:
+
+```json
+"input_map": [
+  {
+    "text_docs": "book.*.chunk.text.*.context"
+  }
+]
+```
+
+Because you're specifying the field to convert into embeddings as a JSON path, you need to set the `full_response_path` to `true` so that the full JSON document is parsed to obtain the input field:
+
+```json
+"full_response_path": true
+```
+
+An ingested document will look as follows. The text in the `context` field will be used to generate embeddings:
+
+```json
+"book": [
+  {
+    "chunk": {
+      "text": [
+        {
+          "chapter": "first chapter",
+          "context": "this is the first part"
+        }
+      ]
+    }
+  }
+]
+```
+
+The Predict API request returns the following response:
+
+```json
+{
+  "inference_results" : [
+    {
+      "output" : [
+        {
+          "name" : "sentence_embedding",
+          "data_type" : "FLOAT32",
+          "shape" : [
+            768
+          ],
+          "data" : [
+            0.25517133,
+            -0.28009856,
+            0.48519906,
+            ...
+          ]
+        }
+      ]
+    }
+  ]
+}
+```
+
+The model records embeddings in the `data` field at the output path `$.inference_results.*.output.*.data`. The `output_map` specifies to record the resulting embeddings in the `context_embedding` field of an ingested document at the same path as the `context` field:
```

Comment:
Can we simplify here as well? The output_map maps the model's embeddings in the path `$.inference_results.*.output.*.data` to the new ingest document field specified by the path `book.*.chunk.text.*.context_embedding`. Also it need not be in the same path as `context`. Depends on user's configuration of the processor